client/components/dt-room-creation.jsx

```diff
@@ -35,21 +38,25 @@ export default class RoomCreation extends React.Component {
       });
   }
 
+  initiateDesktopSocket() {
+    const { endpoint, gameId } = this.state;
+    const desktopSocket = socketIOClient(`${endpoint}/desktop`, { query: `gameId=${gameId}` });
+    desktopSocket.emit('create room', gameId);
+  }
```

Comment:
It's important to store this socket connection so you can disconnected it when this component unmounts. Otherwise, you'll leave a connection floating around in memory.

```js
  insertCodeToDB(event) {
    const req = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ gameId: this.state.gameId })
    };
    fetch('/api/create-game', req)
      .then(response => response.json())
      .then(data => {
        this.socket = socketIOClient(...)
        this.socket.emit('....')
      })
      .catch(error => {
        console.error('Error:', error);
      });
  }

componentWillUnmount() {
  if (this.socket) {
    this.socket.disconnect()
  }
}
```